From 8494004b2a5909e165c0a1745468d03df13b3cbd Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Mon, 21 Jan 2019 16:18:46 -0700
Subject: [PATCH 2/4] sepolicy-legacy: Allow vold to open keymaster firmware

* avc: denied { search } for name="firmware" dev="mmcblk0p23"
  ino=1928 scontext=u:r:vold:s0
  tcontext=u:object_r:firmware_file:s0 tclass=dir permissive=0
* avc: denied { read } for name="keymaster.mdt" dev="mmcblk0p23"
  ino=1950 scontext=u:r:vold:s0
  tcontext=u:object_r:firmware_file:s0 tclass=lnk_file permissive=0
* avc: denied { read } for name="keymaste.mdt" dev="mmcblk0p1" ino=13
  scontext=u:r:vold:s0 tcontext=u:object_r:firmware_file:s0
  tclass=file permissive=0
* avc: denied { open } for name="keymaste.mdt" dev="mmcblk0p1" ino=13
  scontext=u:r:vold:s0 tcontext=u:object_r:firmware_file:s0
  tclass=file permissive=0

Change-Id: Id3dc205a614461a1d174082ea068a77510d423fb
---
 common/vold.te | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/common/vold.te b/common/vold.te
index c64372b..83224ca 100755
--- a/common/vold.te
+++ b/common/vold.te
@@ -12,3 +12,8 @@ wakelock_use(vold)
 allow vold swap_block_device:blk_file r_file_perms;
 
 allow vold persist_file:dir r_dir_perms;
+
+# Allow loading keymaster firmware
+allow vold firmware_file:dir search;
+allow vold firmware_file:file r_file_perms;
+allow vold firmware_file:lnk_file read;
-- 
2.17.1

